Source/WebCore/Scripts/generate-process-sync-data.py

```diff
@@ -54,7 +54,24 @@
 
 
 class SyncedData(object):
-    def __init__(self, name, underlying_type_namespace, underlying_type):
+    def __init__(self, name, underlying_type_namespace, underlying_type, options):
+        self.automatic_location = None
+        self.conditional = None
+        self.header = None
+
+        if options is not None:
+            option_list = options.split()
+            for option in option_list:
+                if option == 'DocumentSyncData':
+                    self.automatic_location = "DocumentSyncData"
+                    continue
+                elif option.startswith('Conditional='):
```

Comment:
It would be nice if this python had some tests that showed the generated cpp with a test .in file, kind of like "make -C Source/WebKit/Scripts/webkit/tests" re-generates cpp from .in files in that directory.